src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_getKeyWorker.kt

```diff
@@ -0,0 +1,145 @@
+@file:Suppress("ClassName")
+
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class BookingResourceIntTest_getKeyWorker : ResourceTest() {
+
+  @Nested
+  inner class Authorisation {
+    @Test
+    fun `returns 401 without an auth token`() {
+      webTestClient.get().uri("/api/bookings/offenderNo/A1234AA/key-worker")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Nested
+    inner class ClientAccess {
+
+      @Test
+      fun `returns 404 when client does not have any roles`() {
+        webTestClient.get().uri("/api/bookings/offenderNo/A1234AA/key-worker")
+          .headers(setClientAuthorisation(listOf()))
+          .exchange()
+          .expectStatus().isNotFound
+          .expectBody().jsonPath("userMessage").isEqualTo("Offender booking with id -1 not found.")
+      }
+
+      @Test
+      fun `returns 404 as ROLE_BANANAS is not override role`() {
```

Comment:
Separated client/user access ready for next change which will mean the return code is different